features/filesystem/fat/FATFileSystem.cpp

```diff
@@ -160,6 +160,26 @@ void ff_memfree(void *p)
 }
 
 // Implementation of diskio functions (see ChaN/diskio.h)
+static WORD disk_get_sector_size(BYTE pdrv)
+{
+    WORD ssize = _ffs[pdrv]->get_erase_size();
+    if (ssize < 512) {
+        ssize = 512;
+    }
+
+    MBED_ASSERT(ssize >= _MIN_SS && ssize <= _MAX_SS);
+    MBED_ASSERT(_ffs[pdrv]->get_read_size() <= _ffs[pdrv]->get_erase_size());
+    MBED_ASSERT(_ffs[pdrv]->get_program_size() <= _ffs[pdrv]->get_erase_size());
+    return ssize;
+}
```

Comment:
get_erase_size() / get_program_size() / disk_get_sector_size() are bit confusing, please verify my understanding below:

1. program size - Can be 1 to 512 bytes (based on device type). Will this be > 512?
2. Erase size - Set as program size (Can we set it as sector size instead) = 512 ?
3. disk_get_sector_size - Here we are forcing it to be 512, why can't we simply set it as 512? Sector size is for file system and not dependent on device.

Reply:
> program size - Can be 1 to 512 bytes (based on device type). Will this be > 512?

It could be > 512 though I don't know of any devices currently that go that high.

> Erase size - Set as program size (Can we set it as sector size instead) = 512 ?

Are you asking about the get_erase_size function inside the driver? The driver doesn't have a concept of sector size. These functions just report the physical limitations of the device. It's up to the filesystem on top to choose a sector size.

> disk_get_sector_size - Here we are forcing it to be 512, why can't we simply set it as 512? Sector size is for file system and not dependent on device.

nor flash specificially needs a sector size of at least 4096 bytes, since it can't erase at <4096 bytes. However, the larger sector size comes with a ram cost, so it's better to use ~512 byte sectors where possible.